resource/item.go

```diff
@@ -24,11 +24,16 @@ type Item struct {
 
 // ItemList represents a list of items
 type ItemList struct {
+	// Results defines the total number of items in the collection matching the current
+	// context without any filter applied. If the storage handler cannot compute this value, -1 is set.
+	Results int
```

Comment:
Please remove it from this PR and let's discuss it in a separate PR.
I'm not ok to change the meaning of `total`, and the distinction between `results` and `total` is not clear. My understanding is that `total` would become the total number of document in the collection, but this is something that would not apply to most backend storage engines. I'm not sure it is relevant to have both, it is more confusing that anything else.

Again, let's discuss that separately.